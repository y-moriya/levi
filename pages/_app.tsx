import { Provider } from 'next-auth/client'
import { QueryClient, QueryClientProvider } from 'react-query'
import { ReactQueryDevtools } from 'react-query/devtools'
import Header from '../components/Header'
import '../styles/global.css'

const queryClient = new QueryClient()
const Layout = ({ children }) => (
  <div className="w-full">
    <Header />
    {children}
  </div>
)

export default function App({ Component, pageProps }) {
  return (
    <QueryClientProvider client={queryClient}>
      <Provider session={pageProps.session}>
        <Layout>
          <Component {...pageProps} />
          <ReactQueryDevtools />
        </Layout>
      </Provider>
    </QueryClientProvider>
  )
}
