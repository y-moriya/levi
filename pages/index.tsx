import PostList from '../components/PostList'
import NewPostForm from '../components/NewPostForm'
import { useSession } from 'next-auth/client'

export const Home = (): JSX.Element => {
  const [session] = useSession()
  return (
    <>
      {!session && <>Please sign in.</>}
      {session && (
        <>
          <NewPostForm />
          <PostList />
        </>
      )}
    </>
  )
}

export default Home
