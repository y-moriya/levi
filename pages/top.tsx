import { useSession } from 'next-auth/client'
import LoopInfoList from '../components/LoopInfoList'

export const Top = (): JSX.Element => {
  const [session] = useSession()
  return (
    <>
      {!session && <>Please sign in.</>}
      {session && (
        <>
          <div className="my-10">
            <LoopInfoList />
          </div>
        </>
      )}
    </>
  )
}

export default Top
