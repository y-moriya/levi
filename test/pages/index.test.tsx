import React from 'react'
import { render } from '../testUtils'
import { Home } from '../../pages/index'
import client from 'next-auth/client'
import { QueryClient, QueryClientProvider } from 'react-query'
import { Session } from 'node-mocks-http'

jest.mock('next-auth/client')

beforeAll(() => {
  const mockSession: Session = {
    expires: '1',
    user: { email: 'euro21st@gmail.com', name: 'Yu MORIYA', image: 'c' },
  }

  ;(client.useSession as jest.Mock).mockReturnValueOnce([mockSession, false])
})

describe('Home page', () => {
  it('matches snapshot', async () => {
    const queryClient = new QueryClient({
      defaultOptions: {
        queries: {},
        mutations: {},
      },
    })

    const { asFragment } = render(
      <QueryClientProvider client={queryClient}>
        <Home />
      </QueryClientProvider>,
      {}
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
