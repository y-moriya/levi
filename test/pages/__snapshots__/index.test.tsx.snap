// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex mx-auto items-center justify-center shadow-lg mt-8 mb-4 max-w-lg"
  >
    <form
      class="w-full max-w-xl bg-white rounded-lg px-4 pt-2"
    >
      <div
        class="flex flex-wrap -mx-3 mb-6"
      >
        <h2
          class="px-4 pt-3 pb-2 text-gray-800 text-lg"
        >
          Add a new Comment
        </h2>
        <div
          class="w-full md-w-full px-3 mb-2 mt-2"
        >
          <textarea
            class="bg-gray-100 rounded border border-gray-400 leading-normal resize-none w-full h-20 py-2 px-3 font-medium placeholder-gray-700 focus:outline-none focus:bg-white"
            id="content"
            name="body"
            placeholder="Type Your Comment"
            required=""
          />
        </div>
        <div
          class="w-full md-w-full flex items-start md:w-full px-3"
        >
          <div
            class="mr-auto"
          />
          <div
            class="-mr-1"
          >
            <button
              class="bg-white text-gray-700 font-medium py-1 px-4 border border-gray-400 rounded-lg tracking-wide mr-1 hover:bg-gray-100"
            >
              Post
            </button>
          </div>
        </div>
      </div>
    </form>
  </div>
  <span>
    Now loading...
  </span>
</DocumentFragment>
`;
